packages/core/upload/admin/src/components/TableList/PreviewCell.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { prefixFileUrlWithBackendUrl, pxToRem } from '@strapi/helper-plugin';
+import { Avatar } from '@strapi/design-system/Avatar';
+import { Flex } from '@strapi/design-system/Flex';
+import { Typography } from '@strapi/design-system/Typography';
+import { Icon } from '@strapi/design-system/Icon';
+import Folder from '@strapi/icons/Folder';
+
+const GenericAssetWrapper = styled(Flex)`
+  span {
+    /* The smallest fontSize in the DS is not small enough in this case */
+    font-size: ${pxToRem(10)};
+  }
+`;
+
+export const PreviewCell = ({ alternativeText, fileExtension, mime, thumbnailURL, type, url }) => {
+  if (type === 'folder') {
+    return (
+      <Flex
+        background="secondary100"
+        height={pxToRem(26)}
+        justifyContent="center"
+        width={pxToRem(26)}
+        borderRadius="50%"
+      >
+        <Icon color="secondary500" as={Folder} />
+      </Flex>
+    );
+  }
+
+  if (mime.includes('image')) {
+    const mediaURL = prefixFileUrlWithBackendUrl(thumbnailURL) ?? prefixFileUrlWithBackendUrl(url);
+
+    return <Avatar src={mediaURL} alt={alternativeText} preview />;
+  }
```

Comment:
Just wondering: in the grid view we render preview images for videos. Does that mean in the list view we don't do that?

Reply:
good question - it wasn't discussed 🤔
since `Avatar` could not handle it right now (we would need a new component), wdyt of discussing it quickly with product and iterate on it after the release?

Reply:
Sounds good to me. Can you create a follow-up issue on Jira? 🚀

Reply:
CONTENT-828 created 🚀